core/templates/domain/objects/RatioObjectFactory.ts

```diff
@@ -0,0 +1,110 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating instances of Ratio
+ * domain objects.
+ */
+
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+
+import { ObjectsDomainConstants } from
+  'domain/objects/objects-domain.constants';
+import { RatioInputAnswer } from
+  'interactions/answer-defs';
+
+export class Ratio {
+  numbers: number[];
+  constructor(numbersList: number[]) {
+    this.numbers = numbersList;
+  }
+
+  toString(): string {
+    return this.numbers.join(':');
+  }
+
+  getNumberOfTerms(): number {
+    return this.numbers.length;
+  }
+  convertToSimplestForm(): number[] {
+    var gcd = (x: number, y: number) => {
+      return y === 0 ? x : gcd(y, x % y);
+    };
+    var gcdResult = this.numbers.reduce(gcd);
+    return this.numbers.map(currentValue => {
+      if (currentValue === 0) {
```

Comment:
You don't actually need the branching here. Just returning currentValue/gcdResult will give you the correct answer.

Reply:
0/0 would lead to undefined value.

Reply:
Then the issue is with gcdResult being 0, right?

Reply:
That's correct also if the result is 4 and we do 0/4 in that case js foes throw an of undefined idky 😅

Reply:
> That's correct also if the result is 4 and we do 0/4 in that case js foes throw an of undefined idky 

I really cannot understand what you are writing, but typing `0/4` in a browser console gives me `0`, as expected...

Reply:
Nevermind it's done now.
I misinterpreted some other test which lead to this conclusion.